### DISCUSSION TITLE
Loop in Homepage

### DISCUSSION THREAD

**Discussion by achlasayyidah** [mecha-cms/mecha#102] [Q&A]
Saya pengguna Blogger dan sebelumnya memang kerap membaca artikel DTE. Melihat DTE migrasi ke Mecha CMS, saya jadi tertarik menggunakan Mecha untuk blog personal saya.

Saya membuat folder sebagai navigasi utama: `jurnal`, `pranala`, `artikel`, `catatan`.

Di empat folder di atas, saya isi tulisan-tulisan yang terkait dengan nama folder tersebut:

```
.\
└── lot\
    └── page\
        ├── jurnal\
        │   ├── jurnal-1.page
        │   ├── jurnal-2.page
        │   └── jurnal-3.page
        ├── pranala\
        │   ├── pranala-1.page
        │   ├── pranala-2.page
        │   └── pranala-3.page
        ├── artikel\
        │   ├── artikel-1.page
        │   ├── artikel-2.page
        │   └── artikel-3.page
        ├── catatan\
        │   ├── catatan-1.page
        │   ├── catatan-2.page
        │   └── catatan-3.page
        ├── jurnal.page
        ├── pranala.page
        ├── artikel.page
        ├── catatan.page
        └── index.page
```

Bisakah saya memanggil semua tulisan tersebut dalam satu `loop` di laman beranda. Kalau di panduannya *kan* saya lihat cuma mengatur halaman beranda dengan menampilkan `loop` dari `.\lot\page\artikel\` saja.

Dalam kasus yang saya tanyakan, saya ingin membuat `loop` berisi berkas-berkas `jurnal-[1-3].page`, `pranala-[1-3].page`, `artikel-[1-3].page`, `catatan-[1-3].page`.

Maaf, saya belum paham cara main di Github, karena baru tahu belakangan ini.

**Comment by taufik-nurrohman** [comment]
> Dalam kasus yang saya tanyakan, saya ingin membuat loop berisi berkas-berkas `jurnal-[1-3].page`, `pranala-[1-3].page`, `artikel-[1-3].page`, `catatan-[1-3].page`.

Itu semua mau dicampur dalam satu pengulangan atau dibuat satu-satu dalam bentuk gawai _recent posts_ di halaman depan? Kalau tujuannya untuk dicampur jadi satu daftar sampai sekarang Saya masih belum menemukan cara yang pas kecuali dengan membuat rute sendiri. Tidak _maintainable_. Kalau semua kategori itu masuk ke dalam anak folder setelah `.\lot\page` masih bisa diusahakan dengan memanfaatkan properti [`deep`](https://mecha-cms.com/store?source=1).

``` .txt
.\
└── lot\
    └── page\
        ├── a\
        │   ├── artikel\
        │   ├── jurnal\
        │   └── pranala\
        └── a.page
```

Di `a.page`:

``` .yaml
---
title: Indeks
deep: 1
...
```

**Reply by achlasayyidah** [reply]
Misalnya saya tetap buat folder seperti pertanyaan awal, dan membuat seperti daftar gawai seperti yang Saudara Taufik utarakan:

``` .html
<!-- loop jurnal -->
<h3>Jurnal</h3>
<ol>
	<li>
		<span>Jurnal 1</span><br>
		<p>Deskripsi jurnal 1.</p>
	</li>
	<!-- etc -->
</ol>

<!-- loop artikel -->
<h3>Artikel</h3>
<ol>
	<li>
		<span>Artikel 1</span><br>
		<p>Deskripsi artikel 1.</p>
	</li>
	<!-- etc -->
</ol>

<!-- etc -->
```

 Dari mana saya harus utak-atik kode Mecha?

Apakah saya harus membuat kondisional di berkas `page.php` di folder `.\lot\layout\page.php`? Seperti di Blogger menggunakan `data:view.isHomepage`?

Ataukah bisa dilakukan dengan cara lain?

**Reply by taufik-nurrohman** [reply]
Pertama-tama buat berkas tema khusus untuk `index.page`:

``` .txt
.\
└── lot\
    └── layout\
        ├── page\
        │   └── index.php
        └── page.php
```

Kemudian di berkas `.\lot\layout\page\index.php` tersebut kita isi markup HTML beserta beberapa gawai _recent posts_. Kurang lebihnya seperti ini:

``` php
<?php

echo '<h2>' . i('Journal') . '</h2>';

foreach (Pages::from(LOT . DS . 'page' . DS . 'jurnal')->sort([-1, 'time'])->chunk(5, 0) as $page) {
    echo '<h3>';
    echo '<a href="' . $page->url . '">';
    echo $page->title;
    echo '</a>';
    echo '</h3>';
    echo '<p>' . $page->description . '</p>';
}

echo '<p><a href="' . $url . '/jurnal/1">' . i('Read More') . '</a></p>';

?>
```

**Reply by achlasayyidah** [reply]
Halo Mas Taufik,

Kalau saya lihat, [DTE](https://dte.web.id/toko) dan [Mecha](https://mecha-cms.com/store) juga menggunakan properti `deep` ya?

Misalnya di DTE:

```
.\
└── lot\
    └── page\
        ├── toko\
        │   ├── informasi\
        │   │   └── tabel-istilah.page
        │   ├── alat\
        │   │   └── generator-kode-warna.page
        │   ├── alat.page
        │   └── informasi.page
        └── toko.page
```

Di tautan di atas `https://dte.web.id/toko`, bisa menampilkan berkas-berkas dalam folder `informasi` dan `alat` di dalam satu `loop` tanpa menyertakan `./toko/informasi` (folder informasi) dan `./toko/alat` (folder alat) sebagai bagian dari `loop`.

Saya mencobanya di localhost, yang terjadi ialah:

`./toko/informasi` (folder informasi) dan `./toko/alat` (folder alat) juga masuk ke dalam `loop` di di laman tautan `./toko`, ada judul dan deskripsinya.

 Bagaimana cara untuk tidak menampilkan `./toko/informasi` (folder informasi) dan `./toko/alat` (folder alat). Jadi, yang tampil di `loop` hanya tautan-tautan berkas di folder informasi dan alat.

Di dalam loop cukup menampilkan:

```
./toko/informasi/tabel-istilah

./toko/alat/generator-kode-warna

dan seterusnya.
```

**Reply by taufik-nurrohman** [reply]
Ganti nama `informasi.page` jadi `informasi.archive`. Halaman arsip tetap akan bisa diakses secara langsung namun tidak akan masuk ke daftar kecuali jika dikondisikan begitu.
